policy/aggregation_type_options.go

```diff
@@ -0,0 +1,607 @@
+	defaultAggregationMinSuffix    = []byte(".lower")
+	defaultAggregationMaxSuffix    = []byte(".upper")
+	defaultAggregationCountSuffix  = []byte(".count")
+	defaultAggregationStdevSuffix  = []byte(".stdev")
+	defaultAggregationMedianSuffix = []byte(".median")
+
+	defaultCounterSuffixOverride = map[AggregationType][]byte{
+		Sum: nil,
+	}
+	defaultTimerSuffixOverride = map[AggregationType][]byte{}
+	defaultGaugeSuffixOverride = map[AggregationType][]byte{
+		Last: nil,
+	}
+)
+
+type options struct {
+	defaultCounterAggregationTypes AggregationTypes
+	defaultTimerAggregationTypes   AggregationTypes
+	defaultGaugeAggregationTypes   AggregationTypes
+	sumSuffix                      []byte
+	sumSqSuffix                    []byte
+	meanSuffix                     []byte
+	lastSuffix                     []byte
+	minSuffix                      []byte
+	maxSuffix                      []byte
+	countSuffix                    []byte
+	stdevSuffix                    []byte
+	medianSuffix                   []byte
+	timerQuantileSuffixFn          QuantileSuffixFn
+	aggTypesPool                   AggregationTypesPool
+	quantilesPool                  pool.FloatsPool
+
+	defaultSuffixes [][]byte
+	counterSuffixes [][]byte
+	timerSuffixes   [][]byte
+	gaugeSuffixes   [][]byte
+
+	counterSuffixOverride map[AggregationType][]byte
+	timerSuffixOverride   map[AggregationType][]byte
+	gaugeSuffixOverride   map[AggregationType][]byte
+
+	defaultCounterAggregationSuffixes [][]byte
+	defaultTimerAggregationSuffixes   [][]byte
+	defaultGaugeAggregationSuffixes   [][]byte
+	timerQuantiles                    []float64
+}
+
+// NewAggregationTypesOptions returns a default AggregationTypesOptions.
+func NewAggregationTypesOptions() AggregationTypesOptions {
+	o := &options{
+		defaultCounterAggregationTypes: defaultDefaultCounterAggregationTypes,
+		defaultGaugeAggregationTypes:   defaultDefaultGaugeAggregationTypes,
+		defaultTimerAggregationTypes:   defaultDefaultTimerAggregationTypes,
+		lastSuffix:                     defaultAggregationLastSuffix,
+		minSuffix:                      defaultAggregationMinSuffix,
+		maxSuffix:                      defaultAggregationMaxSuffix,
+		meanSuffix:                     defaultAggregationMeanSuffix,
+		medianSuffix:                   defaultAggregationMedianSuffix,
+		countSuffix:                    defaultAggregationCountSuffix,
+		sumSuffix:                      defaultAggregationSumSuffix,
+		sumSqSuffix:                    defaultAggregationSumSqSuffix,
+		stdevSuffix:                    defaultAggregationStdevSuffix,
+		timerQuantileSuffixFn:          defaultTimerQuantileSuffixFn,
+		aggTypesPool:                   NewAggregationTypesPool(nil),
+		quantilesPool:                  pool.NewFloatsPool(nil, nil),
+		counterSuffixOverride:          defaultCounterSuffixOverride,
+		timerSuffixOverride:            defaultTimerSuffixOverride,
+		gaugeSuffixOverride:            defaultGaugeSuffixOverride,
+	}
+
+	o.aggTypesPool.Init(
```

Comment:
nit: I usually put the function declaration in the same line as the `Init` call, e.g., 
```
o.aggTypesPool.Init(func() AggregationTypes {
  return ...
})
```
to make it look more succinct and save a few lines.